### ISSUE TITLE
SC2086 on already quoted nested command

### ISSUE THREAD

**Issue by MrKeiKun** [koalaman/shellcheck#2901]
#### For bugs
- Rule Id : SC2086
- My shellcheck version: v0.35.0 VS Code
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

path="some/path"
findArg="! -name 20240101"

readarray -t toBeDeleted <<< "$(find $path -maxdepth 1 -type d $findArg | grep -E '([0-9]{8})' | sed 's\^./\\g' | grep -o "[^/]*$" | sort)"

for k in "${toBeDeleted[@]}"
do
    echo "${path:?}/$k"
    rm -rfv "${path:?}/$k"
done

```

#### Here's what shellcheck currently says:
it says $findArg must be **Double quote to prevent globbing and word splitting.**

#### Here's what I wanted or expected to see:

There should be no error

**Comment by MrKeiKun** [comment]
it seems like path must **double quoted as well** to fixed issue with directory with spaces. but $findArg still gives warning regarding to it.

**Comment by bhundven** [comment]
I get similar with:

```sh
  images="$(find . -name 'Dockerfile*' -type f)"
  for image in "${images[@]}"; do
I   image_dir="$(dirname ${image})" # I: Double quote to prevent globbing and word splitting.
    echo "${image_dir}"
  done
```

`${images}` is already double quoted inside of `"$(dirname ${images}"`.

**Comment by brother** [comment]
@bhundven : You need to quote the variable in the inner shell as well.

`"$(dirname "${image}")"`

**Comment by koalaman** [comment]
ShellCheck is right, the variables in question must indeed be separately quoted. [The wiki](https://github.com/koalaman/shellcheck/wiki/SC2086) explicitly mentions this case.

If it worked as described you could simply have put `echo "$(` as the first line and `)"` as the last line in a file in order to have the whole script run with correct quoting, but alas.
